### ISSUE TITLE
Fatal error generated when insufficient data reached while parsing escaped sequence

### ISSUE THREAD

**Issue by nirosys** [amazon-ion/ion-rust#494]
When insufficient data is encountered while parsing an escaped string sequence (appears to be all types of sequences, but I have not verified), an error is returned that is not correctly identified as Insufficient data. This is causing the raw text reader to fail rather than requesting more data.

**Comment by nirosys** [comment]
After some digging, just re-read the error and realized this is not a UTF-8 boundary issue, but rather the parser not correctly identifying an incomplete error while parsing escaped sequences.

Trivial repro:
```rust
let source = "\"123456\\u269b\"";
//                       ^-- First read stops here.
let mut reader = RawTextReader::new(source.as_bytes()[..10].to_owned());
reader.next()?;
```

Running that returns an Err of:
```
unexpected error after partial escaped sequence: unrecognized input near line 0: could not parse an escaped character (Unicode, hex, or literal): Parsing requires more data: '"123456\u2''
```

**Comment by nirosys** [comment]
A fix for this was merged a while ago, closing.
